core_lib/src/interface/scribblearea.cpp

```diff
@@ -197,11 +200,29 @@ void ScribbleArea::updateFrame(int frame)
         mPixmapCacheKeys.remove(key);
     }
 
-    updateOnionSkinsAround(frame);
+    updateOnionSkinsCacheAround(frame);
 
     update();
 }
 
+void ScribbleArea::updateDirtyFrames()
```

Comment:
You say that but also say that it's important that we update the cache for the frame before and after moving, so I don't think it's an unnecessary step. By all means if you have a better way of doing, do say so, cause I'm running out of ideas.

This way is imo. much better than making it up to the caller to update the frames it's modifying properly.

The name of the function may not be right but the intention is to make it up to scribblearea to update the cache for the frames that has been modified or can be considered dirty on the currently layer.

Reply:
Let me clarify, I think updating the cache for the frame before and after moving is very important and not unnecessary. What I think is unnecessary is having the QList of frames that you are effectively delaying removing them from the cache. Why not just remove them from the cache right away?

Reply:
They are being removed at the next frame update, which should happen right after the move, though it could potentially be delayed. it's the least coupled implementation I have come up with and it does the job. I'm not sure how make it happen right away without involving scribblearea in the proces, do you have any ideas? Maybe if we move the qpixmapcache into the layer model, then I would be able to clear it right away.

Reply:
I think part of my confusion came from not paying attention to what was in Layer vs ScribbleArea. I see now that you don't have access to ScribbleArea in Layer and it indeed wouldn't be good to couple that.

As for alternative approaches, the qpixmapcache is for caching rendered+composited frames, so it would not be good to have that in each layer. It also wouldn't work because it needs access to the output of the CanvasPainter, which Layer does not have. It's a tricky problem. I thought maybe you could make a signal and connect the layers to Editor when they are created/loaded, but looking at the code I see that it's really object that's managing the layers, not layermanager, so that wouldn't really work. I have my worries that we might make some change or come across some corner-case where removeCacheForDirtyFrames does not get called, but your solution might be the best thing short of doing some major architectural changes.